sklearn/cluster/affinity_propagation_.py

```diff
@@ -133,7 +139,10 @@ def affinity_propagation(S, preference=None, convergence_iter=15, max_iter=200,
                     if return_n_iter
                     else (np.array([0]), np.array([0] * n_samples)))
 
-    random_state = np.random.RandomState(0)
+    if random_state is None:
```

Comment:
We have a util for that: `utils.check_random_state`

Reply:
If I am reading the docs right, my if statement should be replaced with utils.check_random_state. @amueller, if that is the case, then do you still want unittests wrapping this function, since they should be caught with unittests from utils.check_random_state? I am good ok with writing those tests, I just want to make sure I following the testing conventions you all use.

Reply:
You used `check_random_state` correctly @rcwoolston :+1: 	

There is no need to check the error message anymore, however a small test making sure that changing the random_state has an effect on the results would still be helpful.

You have some line-too-long issues as can be seen on the `ci/circleci: lint` checker. You can also check yourself with `flake8 path_to_file`